packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/webrtc/PeerConnectionManager.kt

```diff
@@ -372,17 +393,22 @@ internal class PeerConnectionManager(
     qrcMutex.withLock {
       this@PeerConnectionManager.queuedRemoteCandidates = mutableListOf()
     }
+    qlcMutext.withLock {
+      sentSdpOffer = false
+      this.queuedLocalCandidates = mutableListOf()
+    }
     prepareIceServers(integratedTurnServers)
 
     var needsRestart = true
     if (peerConnection == null) {
       setupPeerConnection(localTracks)
       needsRestart = false
     }
+
     peerConnectionMutex.withLock {
       val pc = peerConnection!!
 
-      if (needsRestart) {
+      if (needsRestart && !isExWebrtc) {
```

Comment:
Maybe we can 'calculate' this value inside this function too, instead of keeping it in class context? 🤔